solutions/devsprint-michael-douglas-2/DeliveryApp/Screens/Home/HomeViewController.swift

```diff
@@ -28,17 +28,18 @@ class HomeViewController: UIViewController {
     }
 
     override func viewDidLoad() {
-
+        homeView.updateLoading(with: true)
         navigationController?.navigationBar.prefersLargeTitles = true
-
+        
         deliveryApi.fetchRestaurants { restaurants in
 
             guard let restaurants = restaurants else {
+                self.homeView.updateLoading(with: false)
                 return
             }
 
-            DispatchQueue.main.async {
-
+            DispatchQueue.main.asyncAfter(deadline: .now()+3) {
```

Comment:
Aqui você não precisa adicionar esse delay, você pode chamar direto o `async`.
```suggestion
            DispatchQueue.main.async {
```

Reply:
Esse delay eu coloquei para poder ver o Spiner, por que na hora da build o app já abre muito rapido e já aparece a table view completa, não sei se é cache ou se por que até abrir o app a request ja foi e voltou, mas quando terminar a PR, não vai ter esse delay não.

Reply:
Ahh nice. Realmente a chamada é cacheada, então uma vez que for carregada, ela consome do cache depois. Por isso que não ve o loading, já que é muito rapido o carregamento.